### PR TITLE
Dev/issue 11534 csv export job

### PR DESCRIPTION
**Pull request by mcantelon**
* Use scroll API for export jobs

Fixes issue with export jobs where large result sets can't be retrieved from Elasticsearch without pagination.

### CODE REVIEW

File: lib/job/arRepositoryCsvExportJob.class.php [hunk 1]
```diff
@@ -73,16 +73,17 @@ protected function exportResults($path)
...
 
-    foreach ($resultSet as $hit)
+    // Scroll through results, caching IDs in file, and iterate through IDs in file (given scroll result will expire)
+    foreach ($identifierIteratorFile = arElasticSearchPluginUtil::cacheScrolledSearchResultIdentifiersAsIterableFileObject($search) as $id)
     {
-      if (null === $resource = QubitRepository::getById($hit->getId()))
+      if (null === $resource = QubitRepository::getById($id))
       {
-        $this->error($this->i18n->__('Cannot fetch repository, id: '.$hit->getId()));
+        $this->error($this->i18n->__('Cannot fetch repository, id: '. $id));
```
Comment (jraddaoui) [inline lib/job/arRepositoryCsvExportJob.class.php]: I don't know if we go with spaces around the `.` operator or not, but try not to use both options at the same time ;)

**Review by jraddaoui — COMMENTED** [review]
It looks great @mcantelon!

I just wonder why did you decide to use a temporary file to cache the scroll results instead of our usual cache system. It shouldn't be a huge deal if we're only storing the hit ids.

Thanks!

File: lib/job/arRepositoryCsvExportJob.class.php (hunk 1 above)
Comment (mcantelon) [reply lib/job/arRepositoryCsvExportJob.class.php]: Haha I think I picked that up with Drupal standards or something (no space next to string, but space between var and concatenator). Maybe I'll go with no space as I'm not sure we have a standard.

**Comment by mcantelon** [comment]
I was thinking of avoiding putting all IDs into memory, but I'm probably "prematurely optimizing" so I'll look into using our normal cache system.

File: lib/job/arRepositoryCsvExportJob.class.php (hunk 1 above)
Comment (mcantelon) [reply lib/job/arRepositoryCsvExportJob.class.php]: (Actually I'll use the __ method's sprintf-ish substitution as that's cleaner anyways and as-is it'll mess up translation I think.)

**Review by jraddaoui — APPROVED** [review]
Looks great, thanks Mike!

**Pull request merged**